frame/offences/src/lib.rs

```diff
@@ -54,7 +55,16 @@ pub trait Trait: frame_system::Trait {
 decl_storage! {
 	trait Store for Module<T: Trait> as Offences {
 		/// The primary structure that holds all offence records keyed by report identifiers.
-		Reports get(fn reports): map hasher(blake2_256) ReportIdOf<T> => Option<OffenceDetails<T::AccountId, T::IdentificationTuple>>;
+		Reports get(fn reports): map hasher(blake2_256) ReportIdOf<T>
+			=> Option<OffenceDetails<T::AccountId, T::IdentificationTuple>>;
```

Comment:
I would prefer to keep the `IdentificationTuple` in a separate map if there are multiple reports on the same validator. They are quite large and could bloat the state.

Reply:
yeah this can be separate issue since this is actually old storage and I don't want to do migration as well in the current PR. A follow up can migrate both this and `DeferredOffences` into something that can de-compose `T::IdentificationTuple` into a `Identification` and `ExposureData` (atm this is not possible and not trivial) and then store them as you explained.